subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -933,9 +936,11 @@ public TestFramework testFramework(@Nullable Closure testFrameworkConfigure) {
         //      }
         // }
 
-        if (!userHasConfiguredTestFramework) {
-            testFramework.finalizeValue();
-        }
+        // TODO: Test Framework Selection - Restore this to re-enable fail-fast behavior for Gradle 8
+//        if (!userHasConfiguredTestFramework) {
```

Comment:
As we discussed, we should have a branch that do the necessary changes for 8 ready instead of leaving commented out code here.

Reply:
@ljacomet - Each warning prints the line in the build script where the test framework is set.  Revealing the location was important, to help build authors track down where the problematic configuration was happening, so if they changed the test framework multiple times, they get multiple warnings.

Per the Gradle 8 branch - can I wait to begin that branch from master until after these changes are merged to release, and then release is merged to master?

Reply:
Of course for the Gradle 8 removal branch, wait until all of this is on `master`

Reply:
Tracked here: https://github.com/gradle/gradle/issues/19090